stereo_points/stereo_points.py

```diff
@@ -219,8 +271,22 @@ def onControlPointSelectedChanged(self, newNode):
             newCoordTable.AddNodeReferenceID("stereotaxia_trajLine", newNode.GetID())
             newNode.AddNodeReferenceID("stereotaxia_coordTable", newCoordTable.GetID())
             # add an observer for both nodes, whenever one is renamed, the other one is renamed as well
-            self.lObs_tableNode = newCoordTable.AddObserver(vtk.vtkCommand.ModifiedEvent, self.onCoordTableModified)
-            newNode.AddObserver(vtk.vtkCommand.ModifiedEvent, self.onControlPointNodeModified)
+            self.observers_list.append(
+                [
+                    newCoordTable,
+                    newCoordTable.AddObserver(
+                        vtk.vtkCommand.ModifiedEvent, self.onCoordTableModified
+                    ),
+                ]
+            )
+            self.observers_list.append(
```

Comment:
Do you expect the module to react to all line node changes, or only to the one that is currently selected in the GUI?
Do you want the module to always react, in the background, even when the current module is not active? Then you need to make sure that observers are added whenever suitable nodes are added to the scene (not just when the user goes to this module and interactively changes the node selector).

If you only want the module to react when it is active and only do updates for the currently selected line then you need to remove the old observers when you select a new line.